clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsRequestManager.java

```diff
@@ -256,12 +257,27 @@ public CompletableFuture<Boolean> updateFetchPositions(long deadlineMs) {
                 }
             });
 
+            onMetadataError(metadataError, result);
+
         } catch (Exception e) {
             result.completeExceptionally(maybeWrapAsKafkaException(e));
         }
         return result;
     }
 
+    private void onMetadataError(CompletableFuture<RuntimeException> metadataError, 
+                                 CompletableFuture<Boolean> result) {
+        metadataError.whenComplete((__, error) -> {
+            if (error instanceof AuthorizationException && pendingOffsetFetchEvent != null) {
```

Comment:
With this change we're propagating the metadata error to fail the pending `OffsetFetch` request. But there could be other requests in this manager that need to be aware of the metadata error too I expect, ex. `ListOffsets` (that would be the case of any api call to beginningOffsets/endOffsets/offsetsForTimes without topic access).